dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/user/DatadogUserInfoProviderTest.kt

```diff
@@ -90,6 +90,67 @@ internal class DatadogUserInfoProviderTest {
         ).isEqualTo(customProperties)
     }
 
+    @Test
+    fun `M use immutable properties W setExtraProperties() is called { changing properties values }`(
+        @StringForgery forge: Forge
+    ) {
+        // Given
+        val fakeProperties = forge.exhaustiveAttributes()
+        val fakeExpectedProperties = fakeProperties.toMap()
+        val fakeMutableProperties = fakeProperties.toMutableMap()
+        testedProvider.addUserProperties(fakeMutableProperties)
+
+        // When
+        fakeMutableProperties.keys.forEach {
+            fakeMutableProperties[it] = forge.anAlphabeticalString()
+        }
+
+        // Then
+        assertThat(
+            testedProvider.getUserInfo().additionalProperties
+        ).isEqualTo(fakeExpectedProperties)
+    }
+
+    @Test
+    fun `M use immutable properties W setExtraProperties() is called { adding properties }`(
+        @StringForgery forge: Forge
+    ) {
+        // Given
+        val fakeProperties = forge.exhaustiveAttributes()
+        val fakeExpectedProperties = fakeProperties.toMap()
+        val fakeMutableProperties = fakeProperties.toMutableMap()
+        testedProvider.addUserProperties(fakeMutableProperties)
+
+        // When
+        repeat(forge.anInt(0, 10)) {
+            fakeMutableProperties[forge.anAlphabeticalString()] = forge.anAlphabeticalString()
+        }
+
+        // Then
+        assertThat(
+            testedProvider.getUserInfo().additionalProperties
+        ).isEqualTo(fakeExpectedProperties)
+    }
+
+    @Test
+    fun `M use immutable properties W setExtraProperties() is called { removing properties }`(
+        @StringForgery forge: Forge
+    ) {
+        // Given
+        val fakeProperties = forge.exhaustiveAttributes()
+        val fakeExpectedProperties = fakeProperties.toMap()
+        val fakeMutableProperties = fakeProperties.toMutableMap()
+        testedProvider.addUserProperties(fakeMutableProperties)
+
+        // When
+        fakeMutableProperties.values.drop(forge.anInt(0, fakeMutableProperties.size))
```

Comment:
> [!CAUTION]
> Calling the `values.drop()` doesn't modify the map, only the list of values. You need to call `m.remove(forge.anElementFrom(m.keys))`

> [!NOTE]
> Even if the drop worked, here we would only drop the 'first' n elements. Meaning that you would not remove stuff in the 'middle' of the map. It's not that relevant as Maps aren't ordered, but in some cases it could have some importance.